src/nvim/api/win_config.c

```diff
@@ -1399,6 +1402,14 @@ static bool parse_win_config(win_T *wp, Dict(win_config) *config, WinConfig *fco
     fconfig->_cmdline_offset = (int)config->_cmdline_offset;
   }
 
+  if (HAS_KEY_X(config, statusline)) {
+    if (config->statusline && config->height == 1) {
+      api_set_error(err, kErrorTypeValidation,
+                    "'height' must be greater than 1 when 'statusline' is set");
+      goto fail;
```

Comment:
use `VALIDATE_` macro.
idk the rest of this function does not, but it should (not in this PR)

why is this an error though? a window might temporarily set height=0 and that could just silently not show the statusline?